migration/v2/model/application.go

```diff
@@ -105,3 +108,24 @@ type ImportTag struct {
 	ImportID uint `gorm:"index"`
 	Import   *Import
 }
+
+//
+// Validation Hook to avoid cyclic dependencies.
+func (dep *Dependency) BeforeCreate(db *gorm.DB) (err error) {
+	var nextDeps []*Dependency
+	var nextAppsIDs []uint
+	nextAppsIDs = append(nextAppsIDs, dep.FromID)
+	for len(nextAppsIDs) != 0 {
+		db.Where("ToID IN ?", nextAppsIDs).Find(&nextDeps)
```

Comment:
Amount of SQL queries grows from 1 by "level of depth" of dependency tree (the reason for using ```IN``` query). This looked acceptable for me, but I'm happy to incorporate more efficient solution if recommended one.